TEMPLATE.md

```diff
@@ -0,0 +1,652 @@
+     */
+    public function indexAction()
+    {
+
+    }
+}
+```
+
+`sfaction`
+
+``` php
+public function indexAction()
+{
+    return $this->render('.html.twig');
+}
+```
+
+`sfactiona`
+
+``` php
+/**
+ * @Route("/", name="")
+ * @Template()
+ */
+public function indexAction()
+{
+
+}
+```
+
+`sfem`
+
+``` php
+$em = $this->getDoctrine()->getManager();
+```
+
+`sfrepo`
+
+``` php
+$em->getRepository('Bundle:Repository');
+```
+
+`sfforward`
+
+``` php
+$this->forward('Bundle:Controller:action', array(), array());
+```
+
+`sfredirect`
+
+``` php
+$this->redirect($this->generateUrl('route', array()));
+```
+
+`sfrender`
+
+``` php
+$this->render('Bundle:Folder:template.html.twig', array());
+```
+
+`sfgetsession`
+
+``` php
+$this->getRequest()->getSession();
+```
+
+`sfsetflash`
+
+``` php
+$this->get('session')->getFlashBag()->add('type', 'message');
+```
+
+`sfdump`
+
+``` php
+\Doctrine\Common\Util\Debug::dump();
+```
+
+### Command ###
+
+`sfcommand`
+
+``` php
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class DefaultCommand extends Command
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('default:command')
+            ->setDescription('Default description')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+
+    }
+}
+```
+
+`sfcommandca`
+
+``` php
+use Symfony\Bundle\FrameworkBundle\Command\ContainerAwareCommand;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+
+class DefaultCommand extends ContainerAwareCommand
+{
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $this
+            ->setName('default:command')
+            ->setDescription('Default description')
+        ;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function execute(InputInterface $input, OutputInterface $output)
+    {
+
+    }
+}
+```
+
+### Doctrine ###
+
+#### Classes ####
+
+`sfentityclass`
+
+``` php
+use Doctrine\ORM\Mapping as ORM;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @ORM\Entity(repositoryClass="")
+ * @ORM\Table(name="")
+ */
+class Entity
+{
+    /**
+     * @ORM\Id()
+     * @ORM\Column(type="integer")
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    protected $id;
+}
+```
+
+`sfdocumentclass`
+
+``` php
+use Doctrine\ODM\MongoDB\Mapping\Annotations as MongoDB;
+use Symfony\Component\Validator\Constraints as Assert;
+
+/**
+ * @MongoDB\Document(collection="", repositoryClass="")
+ */
+class Document
+{
+    /**
+     * @MongoDB\Id()
+     */
+    protected $id;
+}
+```
+
+`sfrepository`
+
+``` php
+use Doctrine\ORM\EntityRepository;
+
+class EntityNameRepository extends EntityRepository
+{
+}
+```
+
+`sfgetset`
```

Comment:
Actually it could be better to use the Ctrl+N (Generate) and select `Getters and Setters` because it generates the correct phpdoc based on the class and property. And it's faster :smile: